### ISSUE TITLE
USWDS - Bug: Icon list needs alt text for specific list types

### ISSUE THREAD

**Issue by amycole501** [uswds/uswds#6118]
### Describe the bug

The first example of the [icon list ](https://designsystem.digital.gov/components/icon-list/) shows green checks indicating "yes" and a red x indicating "no" or "don't".

<img width="150" alt="Image" src="https://github.com/user-attachments/assets/ac793c3d-940e-419d-bc05-b77965a73553"/>

The icons on that list are marked as decorative.

We need to add alt text to those icons as they indicate meaning whereas the other icon list examples don't (i.e. those icons are decorative).

### Steps to reproduce the bug

1. Go to https://designsystem.digital.gov/components/icon-list/
2. Listen to the icon lists using JAWS, VO, or NVDA
3. There are no indicators of a "yes" or "no" that reflect the differences in the icons and the meaning the icons imply.

### Expected Behavior

For any icons that convey meaning such as a "yes" or "no" or "do" or "don't" we should add alt text for those indicators.

It may behoove us to also add instructions to that effect on the accessibility section on the page. "Ensure that for any icons that convey meaning, you add relevant alt text"

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://designsystem.digital.gov/about/community/#community-conduct).
- [x] I checked the [current issues](https://github.com/uswds/uswds/issues?q=is%3Aissue+is%3Aopen+label%3A%22Type%3A+Bug%22) for duplicate bug reports.

**Comment by mlloydbixal** [comment]
Next steps is to follow the guidance already outlined in the accessibility section of the icons list component page (just on the list of yes/no icons that are non-decorative"

"If you wish to expose icons to screen readers:
- Provide descriptive text for each icon.
- Remove the aria-hidden="true" attribute and add a aria-labelledby attribute with a value that matches the id of a title element added inside the svg."

And then update the first bullet point to read "Provide descriptive text for each icon that conveys meaning" to include some of what Amy suggested! cc: @amycole501

**Comment by mejiaj** [comment]
@mlloydbixal @amycole501 did we look into alternative solutions and why this approach was chosen?

**Comment by mlloydbixal** [comment]
@mejiaj  I was referencing our own accessibility guidance section for this component and info I found here: https://accessibilityinsights.io/info-examples/web/svg-img-alt/ 

cc @amycole501

**Comment by amycole501** [comment]
A bit more context:
Sometimes SVGs with alt text can cause an accessibility alert to be flagged in automated scanners such as AMP. Especially if someone has tried to add the role=presentation to it.

Overall, I think the labelledby approach coupled with the title element is the most comprehensive way forward. Here's some [info also from W3C on titles](https://www.w3.org/TR/2000/NOTE-SVG-access-20000807/#Equivalent ) 

and here's a really good resource I found  [on accessible SVGs](https://www.unimelb.edu.au/accessibility/techniques/accessible-svgs)  "The SVG has an aria-labelledby attribute which references a title element. Note: the title element should be sufficient on it's own, but we added aria-labelledby to be sure." (edited)

**Comment by thisisdano** [comment]
Maybe it's worth changing the example, since these should be built to work without the icons (and I would say that technically these do — since the list makes just as much sense _with_ the icons as _without_ them). In fact it would be worse if the screen reader read something like "DON'T Avoid large gatherings"

**Comment by amycole501** [comment]
I agree on that specific example being confusing and offering a clearer example of when a person may add meaningful text to an icon list. 

We do say under "accessibility guidance" that the icon's meaning is redundant and we also offer coding guidance on how to add meaningful descriptions of icons if needed. But providing a live example I think, would be helpful.

**Comment by mlloydbixal** [comment]
Came here to say what @amycole501 wrote! The guidance for "meaningful" vs "not meaningful" is on the component's page so I thinking keeping an example of both cases is helpful.

**Comment by annepetersen** [comment]
Making sure we have this issue connected to [the PR](https://github.com/uswds/uswds/pull/6194#top) that I don't think we're going forward with after conversation in [Dev sync](https://docs.google.com/document/d/172TU5BVS9kGZQDuPcAhGoZjaTd2ngzcTayq4WRRTBUs/edit?tab=t.0#heading=h.ze2i8xu3oauw) and [CS/UX sync](https://docs.google.com/document/d/1IAs5HfPXmD3zWRVB5H_ia2HHXMb5yt-gUqJiUectGWQ/edit?tab=t.0#heading=h.75ckzw7gvx40). 

Instead, drafting guidance content defining meaningful vs not meaningful with examples more closely.

**Comment by amycole501** [comment]
I believe this is is now behind handled in [this new icon guidance issue](https://github.com/uswds/uswds-site/issues/2994). @annepetersen can we close this issue out?

**Comment by annepetersen** [comment]
Yep, [as I mentioned over there](https://github.com/uswds/uswds-site/issues/2994#issuecomment-2504533070) 😆
